### PR TITLE
Resolved ssrf vulnerability

### LINKED ISSUE #2650: Fix code scanning alert - Full server-side request forgery
Tracking issue for:
- [ ] https://github.com/OWASP-BLT/BLT/security/code-scanning/118

image

### PR DESCRIPTION
**Pull request by Uttkarsh-raj**
## PR Desc:
This pr is regarding the issue in the server-side request forgery vulnerability. This pr aims to fix https://github.com/OWASP-BLT/BLT/issues/2650 .

### CODE REVIEW

File: website/views.py [hunk 1]
```diff
@@ -3348,6 +3350,18 @@ def get(self, request, *args, **kwargs):
         return render(request, self.template_name, context)
 
 
+def is_valid_url(url):
+    try:
+        parsed_url = urlparse(url)
+        if parsed_url.scheme not in ["http", "https"]:
```
Comment (arkid15r) [inline website/views.py:3356]:
```suggestion
        if parsed_url.scheme not in {"http", "https"}:
```

File: website/views.py [hunk 2]
```diff
@@ -3348,6 +3350,18 @@ def get(self, request, *args, **kwargs):
...
+            return False
+        if re.match(r"^(localhost|127\.|10\.|172\.16\.|192\.168\.)", parsed_url.hostname):
```
Comment (arkid15r) [inline website/views.py:3358]: This looks like an attempt to exclude requests to private network IP addresses. In this case the regex is incomplete, e.g. for `172.16.0.0/12` it won't catch `172.17.*` IPs and so on.

File: website/views.py [hunk 3]
```diff
@@ -156,28 +156,30 @@ def profile_edit(request):
...
-        domain = Domain.objects.get(id=domain)
+        try:
+            domain = Domain.objects.get(id=domain)
+        except Domain.DoesNotExist:
+            return HttpResponseBadRequest("Invalid domain ID")
         company_name = request.POST.get("company")
         company = Company.objects.filter(name=company_name).first()
 
-        if not company:
+        try:
```
Comment (arkid15r) [inline website/views.py:166]: This is very basic check for company name in the response body. Let's at least check that the request isn't being redirected to another location (status code 200 might work)

Comment (Uttkarsh-raj) [reply website/views.py:166]: I have made the changes, plz do check if this satisfies the requirements. Thank you.

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
can you please fix the conflicts?

**Comment by Uttkarsh-raj** [comment]
> can you please fix the conflicts?

On It. Will try to fix this asap.

**Pull request closed without merging**